Danki2/Assets/Scripts/Camera/CustomCamera.cs

```diff
@@ -25,10 +41,22 @@ private void UpdateRotation()
     private void FollowTarget()
     {
         float zDistanceFromFloorIntersect = height / (Mathf.Tan(Mathf.Deg2Rad * angle));
-        float newX = target.transform.position.x;
-        float newY = target.transform.position.y + height;
-        float newZ = target.transform.position.z - zDistanceFromFloorIntersect;
+        var targetPosition = target.transform.position;
 
-        transform.position = new Vector3(newX, newY, newZ);
+        var mouseOffsetH = Input.mousePosition.x / Screen.width - 0.5f;
+        var mouseOffsetV = Input.mousePosition.y / Screen.height - 0.5f;
+
+        desiredPosition.Set(
+            targetPosition.x + mouseOffsetH * mouseFollowFactor,
+            targetPosition.y + height,
+            targetPosition.z - zDistanceFromFloorIntersect + mouseOffsetV * mouseFollowFactor
```

Comment:
Some brackets would be good for the maths here I think. Obviously it's working as intended, but it would make it easier to read